src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -85,9 +87,13 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_mem_reg,
 
     result = cuPointerGetAttribute(&memType, CU_POINTER_ATTRIBUTE_MEMORY_TYPE,
                                    (CUdeviceptr)(address));
-    if ((result == CUDA_SUCCESS) && (memType == CU_MEMORYTYPE_HOST)) {
-        /* memory is allocated with cudaMallocHost which is already registered */
-        *memh_p = NULL;
+    if ((result == CUDA_SUCCESS) && ((memType == CU_MEMORYTYPE_HOST)    ||
+                                     (memType == CU_MEMORYTYPE_UNIFIED) ||
+                                     (memType == CU_MEMORYTYPE_DEVICE))) {
+        /* only host memory not allocated by cuda needs to be registered */
+        /* using deadbeef as VA to avoid gtest error */
+        UCS_STATIC_ASSERT((uint64_t)0xdeadbeef != (uint64_t)UCT_MEM_HANDLE_NULL);
+        *memh_p = (void *)0xdeadbeef;
```

Comment:
why can't you directly assing UCT_MEM_HANDLE_NULL ?

Reply:
I saw some gtests also fail if mem_reg return handle_null